model/host/hostutil.go

```diff
@@ -1038,6 +1046,83 @@ func (h *Host) SpawnHostSetupCommands(settings *evergreen.Settings) (string, err
 	return h.spawnHostSetupConfigDirCommands(conf), nil
 }
 
+// PopulateGithubTokens adds a read-only token for owner/repo associated with the task
+// to h.ProvisionOptions
+func (h *Host) PopulateGithubTokens(ctx context.Context) error {
+	p := h.ProvisionOptions
+	if p.TaskId == "" {
+		return errors.New("missing task id")
+	}
+	if p.FetchOpts == nil {
+		return errors.New("missing fetch opts")
+	}
+	if p.FetchOpts.ProjectOwner == "" || p.FetchOpts.ProjectRepo == "" {
+		return errors.New("missing project owner or project repo")
+	}
+
+	opts := &github.InstallationTokenOptions{
+		Permissions: &github.InstallationPermissions{
+			Contents: utility.ToStringPtr(thirdparty.GithubPermissionRead),
+		},
+	}
+
+	token, err := thirdparty.GetInstallationToken(ctx, p.FetchOpts.ProjectOwner, p.FetchOpts.ProjectRepo, opts)
```

Comment:
nit: it seems like the new thirdparty functions are just wrapping the exact same methods that already exist in the `evergreen` package. I'd advocate strongly for not duplicating functionality into a different package, and just calling the `evergreen` package functions directly.